// @stararc-insurance/core/upload/index.js
import React, { Component } from 'react';
import style from './index.css';
/**
 * 上传
 */
export default class Upload extends Component{
	constructor(props){
		super(props);
		this.state={
			accept:props.accept||'*/*',
			multiple:props.multiple||''
		}
	}
	render(){
		let className= this.props.className?this.props.className:'uploader__input';
		return(
			<div className={style["uploader__box"]}>
				<form action="#" method="post" ref="uploadform" encType="multipart/form-data">
					<input
					multiple={this.state.multiple} 
					ref="uploadInput"
					name="uploadInput"
                    disabled={this.props.disabled}
					className={style[className]} 
					accept={this.state.accept}
					onChange={e=>this.changeHandle(e)}
					type="file"/>
					<button type="button" disabled={this.props.disabled} className={style["uploader__button"]}>{this.props.buttonName}</button>
				</form>
			</div>
		)
	}
	changeHandle(e){
		let value=this.refs.uploadInput;
		let {onChange} = this.props;

		onChange&&onChange();
	}
	getValue(){
		let formEle = this.refs.uploadform;
        var data = new FormData(formEle);

        return data;
	}
}

/**
 * 上传多媒体附件
 */
export  class UploadMedia extends Component{
	constructor(props){
		super(props);
		this.state={
			accept:props.accept,
			multiple:props.multiple||''
		}
	}
	render(){
		return(
			<div className={style["uploader__img--box"]}>
				<form action="#" method="post" ref="uploadform" encType="multipart/form-data">
					<input
					multiple={this.state.multiple} 
					ref="uploadInput"
					name="uploadInput"
                    disabled={this.props.disabled}
					className={style["upload__img"]} 
					accept={this.state.accept}
					onChange={e=>this.changeHandle(e)}
					type="file"/>
					<button type="button" 
						disabled={this.props.disabled} 
						className={style["uploader__img--button"]}>
						{this.props.buttonName}
					</button>
				</form>
			</div>
		)
	}
	changeHandle(e){
		let value=this.refs.uploadInput;
		let {onChange} = this.props;

		onChange&&onChange();
	}
	getValue(){
		let formEle = this.refs.uploadform;
        var data = new FormData(formEle);

        return data;
	}
}


/**
 * 通用的上传
 * 使用是需要在外部控制样式
 * 该组件width,height都为100%，继承父节点
 * eg；
 * <div>
 * 	 <CommonUpload />
 * </div>
 */
export  class CommonUpload extends Component{
	constructor(props){
		super(props);
		this.state={
			accept:props.accept||'*/*',
			multiple:props.multiple||''
		}
	}
	render(){
		let styleCss={
			width:"100%",
			height:"100%"
		};
		// let {styleCss={}} = this.props;
		return(
			<form action="#" style={styleCss} method="post" ref="uploadform" encType="multipart/form-data">
				<input
				multiple={this.state.multiple} 
				ref="uploadInput"
				name="uploadInput"
                disabled={this.props.disabled}
                style={styleCss}
				accept={this.state.accept}
				onChange={e=>this.changeHandle(e)}
				type="file"/>
			</form>
		)
	}
	changeHandle(e){
		let value=this.refs.uploadInput;
		let {onChange} = this.props;

		onChange&&onChange();
	}
	getValue(){
		let formEle = this.refs.uploadform;
        var data = new FormData(formEle);

        return data;
	}
}
