// @stararc-component/table/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.TableTd = exports.TableTr = exports.TableBody = exports.TableHeader = exports.Table = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _index = require('./index.css');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }
// import Tippy from 'tippy';


var Table = exports.Table = function (_Component) {
    _inherits(Table, _Component);

    function Table() {
        _classCallCheck(this, Table);

        return _possibleConstructorReturn(this, (Table.__proto__ || Object.getPrototypeOf(Table)).apply(this, arguments));
    }

    _createClass(Table, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'table',
                { className: _index2.default["common__table"] },
                this.props.children
            );
        }
    }]);

    return Table;
}(_react.Component);

/**
 * 表头显示
 *<TableHeader thArr=["表头th项","表头th项"]/>
 */


var TableHeader = exports.TableHeader = function (_Component2) {
    _inherits(TableHeader, _Component2);

    function TableHeader() {
        _classCallCheck(this, TableHeader);

        return _possibleConstructorReturn(this, (TableHeader.__proto__ || Object.getPrototypeOf(TableHeader)).apply(this, arguments));
    }

    _createClass(TableHeader, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'thead',
                { className: _index2.default["common__table--header"] },
                _react2.default.createElement(
                    'tr',
                    null,
                    this.props.children,
                    this.getTrContent()
                )
            );
        }
    }, {
        key: 'getTrContent',
        value: function getTrContent() {
            var _props$thArr = this.props.thArr;
            var thArr = _props$thArr === undefined ? [] : _props$thArr;


            return thArr.map(function (th, key) {
                return _react2.default.createElement(
                    'th',
                    { key: key, title: th },
                    th
                );
            });
        }
    }]);

    return TableHeader;
}(_react.Component);

/**
 * 表格内容项
 * < TableBody bodyData = [需要显示的表格内容] tdArr = [循环的td字段["field1", "field2", "field3", ]] > < /TableBody>
 */


var TableBody = exports.TableBody = function (_Component3) {
    _inherits(TableBody, _Component3);

    function TableBody() {
        _classCallCheck(this, TableBody);

        return _possibleConstructorReturn(this, (TableBody.__proto__ || Object.getPrototypeOf(TableBody)).apply(this, arguments));
    }

    _createClass(TableBody, [{
        key: 'render',
        value: function render() {

            return _react2.default.createElement(
                'tbody',
                { className: _index2.default["common__table--body"] },
                this.props.children,
                this.getContent()
            );
        }
    }, {
        key: 'getContent',
        value: function getContent() {
            var _this4 = this;

            var _props = this.props;
            var _props$bodyData = _props.bodyData;
            var bodyData = _props$bodyData === undefined ? [] : _props$bodyData;
            var _props$tdArr = _props.tdArr;
            var tdArr = _props$tdArr === undefined ? [] : _props$tdArr;


            return bodyData.map(function (item, key) {
                return _react2.default.createElement(
                    TableTr,
                    _extends({}, self.props, { clickEvent: function clickEvent(id) {
                            _this4.handleClick(item.id);
                        }, key: key, item: item, tdArr: tdArr }),
                    _this4.props.extRows ? _this4.props.extRows[key] : null
                );
            });
        }

        // 行点击事件 确保数据流的行数据中有 id 字段 否则无法触发

    }, {
        key: 'handleClick',
        value: function handleClick(id) {
            var clickEvent = this.props.clickEvent;

            id && clickEvent && clickEvent(id);
        }
    }]);

    return TableBody;
}(_react.Component);

var TableTr = exports.TableTr = function (_Component4) {
    _inherits(TableTr, _Component4);

    function TableTr() {
        _classCallCheck(this, TableTr);

        return _possibleConstructorReturn(this, (TableTr.__proto__ || Object.getPrototypeOf(TableTr)).apply(this, arguments));
    }

    _createClass(TableTr, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'tr',
                { onClick: this.props.clickEvent },
                this.getTdContent(),
                this.props.children
            );
        }
    }, {
        key: 'getTdContent',
        value: function getTdContent() {
            var _this6 = this;

            var _props2 = this.props;
            var _props2$tdArr = _props2.tdArr;
            var tdArr = _props2$tdArr === undefined ? [] : _props2$tdArr;
            var item = _props2.item;


            return tdArr.map(function (td, key) {
                return _react2.default.createElement(
                    'td',
                    { key: key, title: item[td], onClick: _this6.props.onClick },
                    item[td]
                );
            });
        }
    }]);

    return TableTr;
}(_react.Component);

/**
 *  为了让不为零的td项有底线可点击
 */


var TableTd = exports.TableTd = function (_Component5) {
    _inherits(TableTd, _Component5);

    function TableTd() {
        _classCallCheck(this, TableTd);

        return _possibleConstructorReturn(this, (TableTd.__proto__ || Object.getPrototypeOf(TableTd)).apply(this, arguments));
    }

    _createClass(TableTd, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'td',
                { className: _index2.default["td_name"], title: this.props.text },
                _react2.default.createElement('span', { className: _index2.default["bubble__tips"], name: this.props.text || 'nothing',
                    'data-tips': this.props.text }),
                this.props.children
            );
        }
    }, {
        key: 'componnentDidMount',
        value: function componnentDidMount() {
            // new Tippy('.td_name');
        }
    }]);

    return TableTd;
}(_react.Component);
