// @stararc-component/iframe-upload/iframe.upload.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
 * Created by able on 17-4-6.
 */
var IframeUpload = function () {
    function IframeUpload(options) {
        _classCallCheck(this, IframeUpload);

        this.upload_url = options.url;
        this.unique_name = 'unique_name';
        this.upload_button = options.button.cloneNode(true);
        this.onComplete = options.onComplete;
        this.upload();
    }

    /**
     * 上传
     */


    _createClass(IframeUpload, [{
        key: 'upload',
        value: function upload() {
            var body = document.body;
            var iframe = this.create_iframe();
            var form = this.create_form();
            var self = this;

            form.appendChild(this.upload_button);
            form.appendChild(iframe);
            body.appendChild(form);
            iframe.onload = function () {
                var frame_document = this.contentWindow ? this.contentWindow.document : this.contentDocument;
                var response = frame_document.body.innerText;
                if (typeof self.onComplete === 'function') {
                    self.onComplete(JSON.parse(response));
                }
                self.destory_form(form);
            };
            form.submit();
        }

        /**
         * 创建表单
         * @returns {Element}
         */

    }, {
        key: 'create_form',
        value: function create_form() {
            var form = document.createElement('form');
            form.setAttribute('target', this.unique_name);
            form.setAttribute('method', 'POST');
            form.setAttribute('enctype', 'multipart/form-data');
            form.setAttribute('encoding', 'multipart/form-data');
            form.setAttribute('action', this.upload_url);
            form.style.display = 'none';
            return form;
        }

        /**
         * 创建iframe
         * @returns {Element}
         */

    }, {
        key: 'create_iframe',
        value: function create_iframe() {
            var iframe = document.createElement('iframe');
            iframe.src = "javascript:false;";
            iframe.name = this.unique_name;
            iframe.style.display = 'none';
            return iframe;
        }
    }, {
        key: 'get_accept',
        value: function get_accept() {
            var accept = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];

            if (accept.length) {
                return accept.join(';');
            } else {
                return ['image/*', 'video/*', 'audio/*'].join(';');
            }
        }
        /**
         * 移除form
         * @param form
         */

    }, {
        key: 'destory_form',
        value: function destory_form(form) {
            document.removeChild(form);
        }
    }]);

    return IframeUpload;
}();

exports.default = IframeUpload;
