// @stararc-component/header/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

var _index = require('./index.css');

var _index2 = _interopRequireDefault(_index);

var _updatePwd = require('./updatePwd');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var LOGO = require('./img/logo--small.png');

var Header = function (_Component) {
    _inherits(Header, _Component);

    function Header() {
        _classCallCheck(this, Header);

        return _possibleConstructorReturn(this, (Header.__proto__ || Object.getPrototypeOf(Header)).apply(this, arguments));
    }

    _createClass(Header, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'header',
                { className: _index2.default["header"] },
                _react2.default.createElement(
                    'div',
                    { className: _index2.default["header__innerwrap"] },
                    _react2.default.createElement(
                        'div',
                        { className: _index2.default["logo"] },
                        _react2.default.createElement('img', { src: LOGO, alt: '' })
                    ),
                    _react2.default.createElement('div', { className: _index2.default["title__wrap"] }),
                    _react2.default.createElement('div', { className: _index2.default["header__operating"] }),
                    _react2.default.createElement('div', { className: _index2.default["header__right"] })
                )
            );
        }
    }]);

    return Header;
}(_react.Component);

/**
 * 用户登录信息展示
 */


exports.default = Header;

var UserLoginInfo = function (_Component2) {
    _inherits(UserLoginInfo, _Component2);

    function UserLoginInfo() {
        _classCallCheck(this, UserLoginInfo);

        return _possibleConstructorReturn(this, (UserLoginInfo.__proto__ || Object.getPrototypeOf(UserLoginInfo)).apply(this, arguments));
    }

    _createClass(UserLoginInfo, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                { className: _index2.default["operating__user"] },
                '\u6B22\u8FCE\uFF1A\u671D\u9633\u533A\u7BA1\u7406\u5458'
            );
        }
    }]);

    return UserLoginInfo;
}(_react.Component);

/**
 * 用户设置
 */


var UserSetUp = function (_Component3) {
    _inherits(UserSetUp, _Component3);

    function UserSetUp(props) {
        _classCallCheck(this, UserSetUp);

        var _this3 = _possibleConstructorReturn(this, (UserSetUp.__proto__ || Object.getPrototypeOf(UserSetUp)).call(this, props));

        _this3.state = {
            operating: false
        };
        return _this3;
    }

    _createClass(UserSetUp, [{
        key: 'render',
        value: function render() {
            var _this4 = this;

            return _react2.default.createElement(
                'div',
                { className: _index2.default["operating__setUp"], onBlur: function onBlur(e) {
                        return _this4.userSetUpBlur();
                    }, tabIndex: '0' },
                _react2.default.createElement(
                    'div',
                    null,
                    _react2.default.createElement('span', { className: _index2.default["setUp"], onClick: function onClick(e) {
                            return _this4.userSetUpClcik();
                        } }),
                    this.state.operating ? _react2.default.createElement(Operating, { closeUserSetUp: function closeUserSetUp(e) {
                            return _this4.userSetUpBlur();
                        } }) : ''
                )
            );
        }
    }, {
        key: 'userSetUpClcik',
        value: function userSetUpClcik() {
            var operating = !this.state.operating;
            this.setState({
                operating: operating
            });
        }
    }, {
        key: 'userSetUpBlur',
        value: function userSetUpBlur() {
            this.setState({
                operating: false
            });
        }
    }]);

    return UserSetUp;
}(_react.Component);

var Operating = function (_Component4) {
    _inherits(Operating, _Component4);

    function Operating() {
        _classCallCheck(this, Operating);

        return _possibleConstructorReturn(this, (Operating.__proto__ || Object.getPrototypeOf(Operating)).apply(this, arguments));
    }

    _createClass(Operating, [{
        key: 'render',
        value: function render() {
            var _this6 = this;

            return _react2.default.createElement(
                'div',
                { className: _index2.default["operating__content"] },
                _react2.default.createElement(
                    'span',
                    { className: _index2.default["operating_update_pwd"], onClick: function onClick(e) {
                            return _this6.updatePwdClick();
                        } },
                    '\u4FEE\u6539\u5BC6\u7801'
                )
            );
        }
    }, {
        key: 'updatePwdClick',
        value: function updatePwdClick() {
            var closeUserSetUp = this.props.closeUserSetUp;

            closeUserSetUp && closeUserSetUp();
            (0, _updatePwd.updatePassword)();
        }
    }]);

    return Operating;
}(_react.Component);

/**
 * 退回
 */


var UserBack = function (_Component5) {
    _inherits(UserBack, _Component5);

    function UserBack() {
        _classCallCheck(this, UserBack);

        return _possibleConstructorReturn(this, (UserBack.__proto__ || Object.getPrototypeOf(UserBack)).apply(this, arguments));
    }

    _createClass(UserBack, [{
        key: 'render',
        value: function render() {
            var _this8 = this;

            return _react2.default.createElement('div', { className: _index2.default["operating__back"], onClick: function onClick(e) {
                    return _this8.userBack();
                } });
        }
    }, {
        key: 'userBack',
        value: function userBack() {}
    }]);

    return UserBack;
}(_react.Component);

/**
 * 一级菜单导航
 */


var Menu = function (_Component6) {
    _inherits(Menu, _Component6);

    function Menu() {
        _classCallCheck(this, Menu);

        return _possibleConstructorReturn(this, (Menu.__proto__ || Object.getPrototypeOf(Menu)).apply(this, arguments));
    }

    _createClass(Menu, [{
        key: 'render',
        value: function render() {
            var menus = this.getMenu();
            var nav = menus.map(function (m, key) {
                return _react2.default.createElement(
                    _reactRouter.Link,
                    { key: key, to: m.path, className: _index2.default["menu__text"],
                        activeClassName: _index2.default["menu__text--hover"] },
                    m.name
                );
            });
            return _react2.default.createElement(
                'div',
                { className: _index2.default["menu"] },
                nav
            );
        }
    }, {
        key: 'getMenu',
        value: function getMenu() {
            return [{
                name: '统计',
                path: '/statistic'
            }];
        }
    }]);

    return Menu;
}(_react.Component);
