// @stararc-component/header/src/index.js
import React, {Component} from 'react';
import {Link} from 'react-router';
import style from './index.css';

import {convertCookieToObj} from "@stararc-insurance/help-tools";

const LOGO = require('./img/logo--small.png');

import {updatePassword} from './updatePwd';
export default class Header extends Component {
    render() {
        return (
            <header className={style["header"]}>
                <div className={style["header__innerwrap"]}>
                    <div className={style["logo"]}>
                        <img src={LOGO} alt=""/>
                    </div>
                    <div className={style["title__wrap"]}>
                        {/*<div className={style["logo__title"]}>
                            <div>安全生产大数据分系统</div>
                        </div>*/}
                    </div>
                    <div className={style["header__operating"]}>
                        <UserLoginInfo/>
                        {/*<UserSetUp/>*/}
                        <UserBack loginOut={this.props.loginOut} linkUrl={this.props.url}/>
                    </div>
                    <div className={style["header__right"]}>
                        {/*<Menu />*/}
                    </div>
                </div>
            </header>
        )
    }
}

/**
 * 用户登录信息展示
 */
class UserLoginInfo extends Component{
    render(){
        let infor = convertCookieToObj();

        return(
            <div className={style["operating__user"]}>欢迎：{infor.user_name}</div>
        )
    }
}

/**
 * 用户设置
 */
class UserSetUp extends Component{
    constructor(props){
        super(props)
        this.state={
            operating:false
        }
    }
    render(){
        return(
            <div className={style["operating__setUp"]} onBlur={e=>this.userSetUpBlur()} tabIndex="0">
                <div>
                    <span className={style["setUp"]} onClick={e=>this.userSetUpClcik()} >
                    </span>
                    {this.state.operating?<Operating closeUserSetUp={e=>this.userSetUpBlur()}/>:''}
                </div>
            </div>

        )
    }
    userSetUpClcik(){
        let operating=!this.state.operating;
        this.setState({
            operating:operating
        })
    }
    userSetUpBlur(){
        this.setState({
            operating:false
        })
    }
}

class Operating extends Component{
    render(){
        return(
            <div className={style["operating__content"]}>
                <span className={style["operating_update_pwd"]} onClick={e=>this.updatePwdClick()}>修改密码</span>
            </div>
        )
    }
    updatePwdClick(){
        let {closeUserSetUp}=this.props;
        closeUserSetUp&&closeUserSetUp();
        updatePassword();
    }
}




/**
 * 退回
 */
class UserBack extends Component{
    render(){
        return(
            <a className={style["operating__back"]} href={this.props.linkUrl} onClick={e=>this.login_out()}></a>
        )
    }
    login_out(){
        // let {loginOut} = this.props;

        // loginOut && loginOut();
    }
}

/**
 * 一级菜单导航
 */
class Menu extends Component {
    render() {
        let menus = this.getMenu();
        let nav = menus.map(function (m, key) {
            return <Link key={key} to={m.path} className={style["menu__text"]}
                    activeClassName={style["menu__text--hover"]}>{m.name}</Link>;
        })
        return (
            <div className={style["menu"]}>
                {nav}
            </div>
        );
    }

    getMenu() {
        return [{
            name: '统计',
            path: '/statistic'
        }]
    }    
}
