// @stararc-component/input/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index.css');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Input = function (_Component) {
    _inherits(Input, _Component);

    function Input(props) {
        _classCallCheck(this, Input);

        var _this = _possibleConstructorReturn(this, (Input.__proto__ || Object.getPrototypeOf(Input)).call(this, props));

        _this.state = { value: props.defaultValue || '' };
        return _this;
    }

    _createClass(Input, [{
        key: 'render',
        value: function render() {
            var _this2 = this;

            var _props = this.props;
            var className = _props.className;
            var type = _props.type;
            var placeholder = _props.placeholder;
            var _props$styleCss = _props.styleCss;
            var styleCss = _props$styleCss === undefined ? {} : _props$styleCss;
            var maxLength = _props.maxLength;


            className = className ? className : "input";

            return _react2.default.createElement('input', {
                ref: 'input',
                style: styleCss,
                disabled: this.props.disabled,
                className: _index2.default[className],
                readOnly: this.props.readonly,
                value: this.state.value,
                onFocus: function onFocus(e) {
                    _this2.focusHandle(e);
                },
                onBlur: function onBlur(e) {
                    _this2.blurHandle(e);
                },
                onChange: function onChange(e) {
                    _this2.changeHandle(e);
                },
                onKeyUp: function onKeyUp(e) {
                    return _this2.keyPressHandle(e);
                },
                maxLength: maxLength,
                type: type,
                placeholder: placeholder });
        }
    }, {
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            if (nextProps.defaultValue != this.props.defaultValue) {
                this.setState({ value: nextProps.defaultValue || "" });
            }
        }
    }, {
        key: 'focusHandle',
        value: function focusHandle(e) {
            var onFocus = this.props.onFocus;

            onFocus && onFocus(e);
        }
    }, {
        key: 'blurHandle',
        value: function blurHandle(e) {
            var onBlur = this.props.onBlur;

            onBlur && onBlur(e);
        }
    }, {
        key: 'changeHandle',
        value: function changeHandle(e) {
            var value = e.target.value;
            var onChange = this.props.onChange;

            this.setState({ value: value }, function () {
                onChange && onChange(e);
            });
        }
    }, {
        key: 'keyPressHandle',
        value: function keyPressHandle(e) {
            var value = e.target.value;
            var onKeyup = this.props.onKeyup;

            this.setState({ value: value }, function () {
                onKeyup && onKeyup(e);
            });
        }
    }, {
        key: 'getFocus',
        value: function getFocus() {
            this.refs.input.focus();
        }
    }, {
        key: 'blurFocus',
        value: function blurFocus() {
            this.refs.input.blur();
        }
    }, {
        key: 'getValue',
        value: function getValue() {
            return this.state.value;
        }
    }]);

    return Input;
}(_react.Component);

exports.default = Input;
