// wheel-datepicker/sub/Date.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _DatePicker = require('../DatePicker.css');

var _DatePicker2 = _interopRequireDefault(_DatePicker);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by able on 2017/1/5 0005.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var DatePickerDate = function (_Component) {
    _inherits(DatePickerDate, _Component);

    function DatePickerDate() {
        _classCallCheck(this, DatePickerDate);

        return _possibleConstructorReturn(this, (DatePickerDate.__proto__ || Object.getPrototypeOf(DatePickerDate)).apply(this, arguments));
    }

    _createClass(DatePickerDate, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                { className: _DatePicker2.default["datepicker__content"] },
                _react2.default.createElement(DatePickerHeader, this.props),
                _react2.default.createElement(DatePickerWeek, { weekLabels: this.props.conf.weekLabels }),
                _react2.default.createElement(DatePickerMonth, this.props)
            );
        }
    }]);

    return DatePickerDate;
}(_react.Component);

exports.default = DatePickerDate;

var DatePickerWeek = function (_Component2) {
    _inherits(DatePickerWeek, _Component2);

    function DatePickerWeek() {
        _classCallCheck(this, DatePickerWeek);

        return _possibleConstructorReturn(this, (DatePickerWeek.__proto__ || Object.getPrototypeOf(DatePickerWeek)).apply(this, arguments));
    }

    _createClass(DatePickerWeek, [{
        key: 'render',
        value: function render() {
            var week = this.props.weekLabels.map(function (w, index) {
                return _react2.default.createElement(
                    'span',
                    { key: index, className: _DatePicker2.default["weekLabel"] },
                    w
                );
            });
            return _react2.default.createElement(
                'div',
                { className: _DatePicker2.default["datepicker__week"] },
                week
            );
        }
    }]);

    return DatePickerWeek;
}(_react.Component);

var DatePickerMonth = function (_Component3) {
    _inherits(DatePickerMonth, _Component3);

    function DatePickerMonth() {
        _classCallCheck(this, DatePickerMonth);

        return _possibleConstructorReturn(this, (DatePickerMonth.__proto__ || Object.getPrototypeOf(DatePickerMonth)).apply(this, arguments));
    }

    _createClass(DatePickerMonth, [{
        key: 'render',
        value: function render() {
            var _props$conf = this.props.conf;
            var today = _props$conf.today;
            var isCurrentMonth = _props$conf.isCurrentMonth;
            var monthLabels = _props$conf.monthLabels;
            var selectedDateObj = _props$conf.selectedDateObj;
            var currentDateObj = _props$conf.currentDateObj;

            var self = this;
            var isEq = currentDateObj.year === selectedDateObj.year && currentDateObj.month === selectedDateObj.month;
            var month = monthLabels.map(function (m, index) {
                var className = isEq && m === selectedDateObj.date ? 'date--selected' : 'monthLabel';
                if (isCurrentMonth && m === today) {
                    className = selectedDateObj.date === today ? 'date--selected' : 'date--today';
                }
                return m === '' ? _react2.default.createElement('div', { key: index, className: _DatePicker2.default['date--disabled'] }) : _react2.default.createElement(
                    'div',
                    { key: index, className: _DatePicker2.default[className], onClick: function onClick(e) {
                            return self.clickHandle(m);
                        } },
                    _react2.default.createElement(
                        'span',
                        { className: _DatePicker2.default["date"] },
                        m
                    ),
                    _react2.default.createElement('span', { className: _DatePicker2.default["date__animation"] })
                );
            });
            return _react2.default.createElement(
                'div',
                { className: _DatePicker2.default["monthLabel__wrap"] },
                month
            );
        }
    }, {
        key: 'clickHandle',
        value: function clickHandle(m) {
            this.setState({ selectedIndex: m });
            this.props.onSelect(m);
        }
    }]);

    return DatePickerMonth;
}(_react.Component);

var DatePickerHeader = function (_Component4) {
    _inherits(DatePickerHeader, _Component4);

    function DatePickerHeader() {
        _classCallCheck(this, DatePickerHeader);

        return _possibleConstructorReturn(this, (DatePickerHeader.__proto__ || Object.getPrototypeOf(DatePickerHeader)).apply(this, arguments));
    }

    _createClass(DatePickerHeader, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                { className: _DatePicker2.default["datepicker__header"] },
                _react2.default.createElement(
                    'span',
                    { className: _DatePicker2.default["month__prev"], onClick: this.props.onTurnPrev },
                    '< '
                ),
                _react2.default.createElement(
                    'span',
                    { className: _DatePicker2.default["datetext"] },
                    this.props.conf.currentMonth
                ),
                _react2.default.createElement(
                    'span',
                    { className: _DatePicker2.default["month__next"], onClick: this.props.onTurnNext },
                    ' >'
                )
            );
        }
    }]);

    return DatePickerHeader;
}(_react.Component);
